libs/banglejs/jswrap_bangle.c

```diff
@@ -5073,27 +5073,38 @@ JsVar *jswrap_banglejs_getPressure() {
   /* Occasionally on some devices (https://github.com/espruino/Espruino/issues/2137)
   you can get an I2C error. This stops the error from being fired when getPressure
   is called and instead rejects the promise. */
+  JsVar *exception = jspGetException();
+  if (exception) {
+    jspromise_reject(promisePressure, exception);
+    jsvUnLock2(promisePressure, exception);
+    JsVar *r = promisePressure;
+    promisePressure = 0;
+    return r;
+  }
+
   bool hadError = jspHasError();
   if (hadError) {
-    JsVar *exception = jspGetException();
+    JsVar *msg = jsvNewFromString("I2C barometer error");
+    JsVar *exception = jswrap_internalerror_constructor(msg);
```

Comment:
> ```
> libs/banglejs/jswrap_bangle.c:5088:24: error: implicit declaration of function 'jswrap_internalerror_constructor'; did you mean 'jswrap_typedarray_constructor'? [-Werror=implicit-function-declaration]
> ```

I could include the header, but want to confirm - is this expected or should I be going through another function for this?

Reply:
I'd add that to the headers - I think it's ok. Usually we'd create an exception with `jsExceptionHere` but that actually fires off the exception, which we don't want to do, so I think your method is fine

Reply:
Agreed - I've included the header, all sorted